File: SupportService/Domain/Entities/Comment.cs
namespace Domain.Entities;

public class Comment
{
    public int Id { get; set; }

    public string Message { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    #region Navigation Properties

    public int? ServiceRequestId { get; set; }

    public ServiceRequest ServiceRequest { get; set; }

    public int? CreatedById { get; set; }

    public User CreatedBy { get; set; }

    public List<Attachment> Attachments { get; set; }

    #endregion
}

File: SupportServiceUi/src/app/interfaces/Comment.ts
import { AttachmentPreview } from "./Attachment"

export interface Comment {
    id: number,
    message: string,
    name: string,
    email: string,
    createdAt: Date,
    attachments: AttachmentPreview[]
}

export interface CommentToCreate {
    message: string,
    createdById: number,
    serviceRequestId: number
}